hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -256,6 +256,8 @@ public SessionImpl(SessionFactoryImpl factory, SessionCreationOptions options) {
 		// NOTE : pulse() already handles auto-join-ability correctly
 		getTransactionCoordinator().pulse();
 
+		getSession().setHibernateFlushMode( ConfigurationHelper.getFlushMode( getSessionProperty( AvailableSettings.FLUSH_MODE ), FlushMode.AUTO ) );
+
```

Comment:
@Sanne should we add the `FlushMode` to the `FastSessionServices`?

Reply:
Actually, `FastSessionServices` contains `FlushMode` default value. If you drill down the `getSessionProperty()` you can see it will look up in `FastSessionServices` already:
<img width="741" alt="Screen Shot 2020-01-27 at 5 34 12 PM" src="https://user-images.githubusercontent.com/8211458/73220006-6b6e2080-412b-11ea-84e9-0a7513b593a0.png">

Reply:
Right, looks good. Is something not working as expected @dreab8 ? Feel free to add tests :)

Reply:
I haven't noticed that `getSessionProperty` takes the value from the `FastSessionServices`

Reply:
hum both `ConfigurationHelper#getFlushMode` and the `getSessionProperty` methods are rather slow though. I see what @dreab8 meant now, I will add an additional fix to this.